### PR TITLE
[VPlan] Implementation of onlyFirstLaneUsed for VPLiveOut class

### PR DESCRIPTION
**Pull request by arcbbb**
Following up on #83068, when scalarizing VPWidenPointerInductionRecipe,
The `onlyScalarsGenerated` checks whether VF is scalable. With scalable
VF, it requires all user to use the first lane only.
However if any user happens to be VPLiveOut, the check inevitably fails.

This patch addresses this by implementing onlyFirstLaneUsed for the
VPLiveOut class.
It ensures that if the operand is a VPWidenPointerInductionRecipe, it
returns true.

### CODE REVIEW

File: llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp [hunk 1]
```diff
@@ -186,6 +186,13 @@ bool VPRecipeBase::mayHaveSideEffects() const {
   }
 }
 
+bool VPLiveOut::onlyFirstLaneUsed(const VPValue *Op) const {
+  assert(is_contained(operands(), Op) && "Op must be an operand of the recipe");
+
+  return vputils::isUniformAfterVectorization(getOperand(0)) ||
+         isa<VPWidenPointerInductionRecipe>(Op);
```
Comment (arcbbb) [inline llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp:193]: fixupIVUsers removes the VPLiveOut after its fix. Given that the end value of IV is derived from TripCount, I just think we might alternatively eliminate  the use of ptr-iv by  creating a new VPValue for VPLiveOut.

**Comment by arcbbb** [comment]
ping

**Comment by fhahn** [comment]
Is this still needed?

**Comment by arcbbb** [comment]
> Is this still needed?

No, after commit fc9cd3272b50f4ee9f18c4ab82c278bbb014d99f `[VPlan] Don't add live-outs for IV phis` the need is gone. Thanks!

**Pull request closed without merging**